mobile/src/main/java/net/activitywatch/android/watcher/UsageStatsWatcher.kt

```diff
@@ -156,6 +158,13 @@ class UsageStatsWatcher constructor(val context: Context) {
                 val event = UsageEvents.Event()
                 usageEvents.getNextEvent(event)
                 if(event.eventType !in arrayListOf(UsageEvents.Event.ACTIVITY_RESUMED, UsageEvents.Event.ACTIVITY_PAUSED, UsageEvents.Event.SCREEN_INTERACTIVE, UsageEvents.Event.SCREEN_NON_INTERACTIVE)) {
+                    if(event.eventType == UsageEvents.Event.KEYGUARD_HIDDEN){
+                        val timestamp = DateTimeUtils.toInstant(java.util.Date(event.timeStamp))
+                        // pulse time is 1.0 because getLastEventTime() returns last time of an activity event,
+                        // and if an unlock happens after last activity, it will get sent twice.
+                        // Setting pulsetime 1.0 will merge those events together
```

Comment:
I wrote the comment as a side-note for the following scenario: Currently what's happening is that UsageStatsWatcher queries only the period starting from the last event in the aw-android-test bucket till the present. Let's say activity event B(opening Chrome, closing Settings app, etc) is the last Android Activity that AndroidStats recorded. Now if an unlock event A happens after an activity event B, both will be stored in the correct buckets. Next time when you query Android Statistics, you query is based on event B, so your Android Stats  include unlock event A again and will add it again to the unlock bucket. My comment was saying that this shouldn't be a problem, because events that happen at the same time will get merged together in the bucket.

I also realized that they will get merged together even if pulse-time is 0(correct me if i'm wrong) since they occur at same time, so i'll change pulse-time to 0 ( which will also count unlocks that happen 1 second apart as separate, which is intended behaviour).

Reply:
Oh now I get what you mean, good point. If it's the same event (has the exact same timestamp and data) a heartbeat with pulsetime should indeed merge. I don't think we have any tests for that though, so I can not guarantee it but that's the idea of how it should work at least.